### PR TITLE
dev-python/gevent: IUSE=events for zope-event

### PR DESCRIPTION
**Pull request by tgbugs**
The gevents.events module depends on zope.event at runtime. This commit
adds an events use flag which adds dev-python/zope-event as a dependency.

Signed-off-by: Tom Gillespie <tgbugs@gmail.com>

### CODE REVIEW

File: dev-python/gevent/metadata.xml [hunk 1]
```diff
@@ -5,6 +5,9 @@
     <email>python@gentoo.org</email>
     <name>Python</name>
   </maintainer>
+  <use>
+    <flag name="events">Add dev-python/zope-event as a runtime dependency.</flag>
```
Comment (mgorny) [inline dev-python/gevent/metadata.xml:9]: The flag description should really say what it does, i.e. how would the user benefit from enabling it. Also `<pkg/>` around package name.

Comment (tgbugs) [reply dev-python/gevent/metadata.xml]: Updated with a hopefully more informative description. I expect that this flag is likely to be used by other packages that depend on the gevent events system and want to set that as a required use flag (or at least, that is my use case).

**Comment by tgbugs** [comment]
I don't know whether you want the bump to 20.6.2 in this pr but I saw that it had been released while checking on deps, so I pulled it in.

**Comment by tgbugs** [comment]
I assume you mean depend on `dev-python/zope-event` directly which I understand from a gentoo point of view. However, upstream also lists events as an extra so this use flag mirrors the behavior of setup.py, and thus for any developer who might use gevent having the use flag is a convenience to mimic the upstream behavior. Does this make sense?

**Comment by mgorny** [comment]
I mean having the flag on the package is fine, just that other packages shouldn't depend on the flag.

File: dev-python/gevent/Manifest [hunk 2]
```diff
@@ -3,3 +3,4 @@ DIST gevent-20.5.1.tar.gz 5566653 BLAKE2B 1d9a474681573ec2e9d3c0d40fbae8a91b8cfb
...
 DIST gevent-20.6.0.tar.gz 5803237 BLAKE2B 97b988964ef80fcc1c3238718b6802016fb46742c94e7e56b5196139181ff82ce60816e0a8ccc5c3e1c3b73ec11cc705884b5ac5524dbd5116dc0d8722ccc7a4 SHA512 28ee030934d3c379577c9f86b76838e5d76c7ea7c0880cdeac9dc5ead497ae0600fb02b2984c7312fd42319142200c674b75da2e93b58fd4a99882d601a95108
 DIST gevent-20.6.1.tar.gz 5806150 BLAKE2B fb75e440f31ad2952d1395780a66ee673c55e62b635a2c6757efe9706c404d126c38a613ac08cd088900c0b10913fe9ce9d57cec8a7cb252ae87912142cfafce SHA512 b7ceb5251b01972a11e3d51e1ebcaaca7c08d36b6ba7b138bee57f0d5da4553539a0ad538e8a433321bdac396788acbfb0ac294191e829df3f066d00816310bb
+DIST gevent-20.6.2.tar.gz 5806612 BLAKE2B 8b1fbd991cb169292d33001bb0df06702d36eb3849c9251e4fac9e2fb5826eda3435971e39e209f1f537536081ce2730e561e91c398ee66a9b1d55c9d0e5a81a SHA512 65caa9caf8a2913b4c3bcc9a303169ce5837a0d7ccf4fb6b7583bf4ec3a0b288d65fc7e28399f0715e61aed128ffc6e67fc9b455474e47263236136a492021f4
```
Comment (mgorny) [inline dev-python/gevent/Manifest:6]: This doesn't seem to belong to this commit.

File: dev-python/gevent/metadata.xml [hunk 3]
```diff
@@ -6,7 +6,7 @@
     <name>Python</name>
   </maintainer>
   <use>
-    <flag name="events">Add dev-python/zope-event as a runtime dependency.</flag>
+    <flag name="events">Enables the gevent events system which is required by libraries that use gevent monkey patching. This adds <pkg>dev-python/zope-event</pkg> and <pkg>dev-python/zope-interface</pkg> as runtime dependencies.</flag>
```
Comment (mgorny) [inline dev-python/gevent/metadata.xml:9]: This neither.

**Comment by tgbugs** [comment]
Ah, I understand.

File: dev-python/gevent/metadata.xml (hunk 3 above)
Comment (tgbugs) [reply dev-python/gevent/metadata.xml:9]: Will move to its own commit.

File: dev-python/gevent/Manifest (hunk 2 above)
Comment (tgbugs) [reply dev-python/gevent/Manifest:6]: Oops yes, that should have been with the version bump commit, will fix and rebase.

**Comment by tgbugs** [comment]
I see that these deps have been added by default in the latest version. Should I close this PR or is this use flag still useful?

**Comment by mgorny** [comment]
I'm sorry, I've forgotten about this. Will merge it today.

**Comment by mgorny** [comment]
Merged as 86abeeb2136bf8961a234de44bb2bab7e6d2d06a.
